### PR TITLE
Spine v3.8.0 Fix missing interface bindings

### PR DESCRIPTION
**Pull request by Canvasfull**
Re: #

### Changelog

1. Add bone & IkConstraintData & IkConstraint & PathConstraintData & PathConstraint setter bindings.
2. Add TrackEntry & AnimationState & Animation setter bindings.
3. Fix the problem that SkeletonData removal skin and animation are not removed on the editor.
4. Fix an issue that caused the skeleton animation before the render list to render incorrectly after disabling the node in the editor.
5. add Skeleton getState interface.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/spine/skeleton-system.ts [hunk 1]
```diff
@@ -70,6 +71,10 @@ export class SkeletonSystem extends System {
         if (this._skeletons.has(skeleton)) {
             this._skeletons.delete(skeleton);
         }
+        if(!EDITOR_NOT_IN_PREVIEW) return;
+        this._skeletons.forEach((skeleton) => {
```
Comment (SantyWang) [inline cocos/spine/skeleton-system.ts]: It's strange! why need this?

File: cocos/spine/skeleton.ts [hunk 2]
```diff
@@ -600,6 +600,16 @@ export class Skeleton extends UIRenderer {
     public onRestore () {
 
     }
+    /**
+     * @en Gets the animation state object.
+     * @zh 获取动画状态。
+     * @method getState
+     * @return {sp.spine.AnimationState} state
+     */
+    public getState(): spine.AnimationState | undefined {
```
Comment (SantyWang) [inline cocos/spine/skeleton.ts:609]: Why export this state?

File: native/cocos/editor-support/spine-wasm/spine-type-export.cpp [hunk 3]
```diff
@@ -583,7 +612,29 @@ EMSCRIPTEN_BINDINGS(spine) {
         //.function("localToWorld", &Bone::localToWorld)
         .function("worldToLocalRotation", &Bone::worldToLocalRotation)
         .function("localToWorldRotation", &Bone::localToWorldRotation)
-        .function("rotateWorld", &Bone::rotateWorld);
+        .function("rotateWorld", &Bone::rotateWorld)
+        .function("setX", &Bone::setX)
```
Comment (SantyWang) [inline native/cocos/editor-support/spine-wasm/spine-type-export.cpp:616]: Need to verify these interfaces.

File: cocos/spine/skeleton.ts (hunk 2 above)
Comment (Canvasfull) [reply cocos/spine/skeleton.ts:609]: This is the interface already exposed by 3.7.3 and in use by users.
Screenshot 2023-07-05 at 14 08 49

File: native/cocos/editor-support/spine-wasm/spine-type-export.cpp (hunk 3 above)
Comment (Canvasfull) [reply native/cocos/editor-support/spine-wasm/spine-type-export.cpp:616]: Bone data is validated at the time of addition.

File: cocos/spine/skeleton-system.ts (hunk 1 above)
Comment (Canvasfull) [reply cocos/spine/skeleton-system.ts]: Not refreshing will cause the spine to render the animation incorrectly.

File: native/cocos/editor-support/spine/SlotData.h [hunk 4]
```diff
@@ -76,6 +76,8 @@ class SP_API SlotData : public SpineObject {
 
     int getIndex();
 
+    int hash;
```
Comment (minggo) [inline native/cocos/editor-support/spine/SlotData.h:79]: Why use public member variable?

File: native/cocos/editor-support/spine-wasm/spine-wasm.cpp [hunk 5]
```diff
@@ -44,31 +45,72 @@ SkeletonData* SpineWasmUtil::querySpineSkeletonDataByUUID(const std::string& uui
...
+                slotArray[i]->hash = itValue + 10000;
+            }else{
+                slotArray[i]->hash = 30000;
+            }
+        }
+    }
+}
+
+
+
```
Comment (minggo) [inline native/cocos/editor-support/spine-wasm/spine-wasm.cpp]: extra space

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

File: native/cocos/editor-support/spine/SlotData.h (hunk 4 above)
Comment (Canvasfull) [reply native/cocos/editor-support/spine/SlotData.h:79]: When updating the render data, you need to record the texture index value corresponding to the slot. So a public property is used here.
Screenshot 2023-07-07 at 16 00 38

Comment (SantyWang) [reply native/cocos/editor-support/spine/SlotData.h:79]: Hack for this version, will be removed in next version.

**Pull request merged**